lib/foreman/silenced_logger.rb

```diff
@@ -1,15 +1,38 @@
 # Wrap a ::Logging::Logger and implement #silence to temporarily increase the
 # min log level to execute a block of code. Used by sprockets-rails' quiet
-# assets logging feature.
+# assets logging feature. Foreman uses logging gem instead Rails logging stack
+# and it does not provide silencing feature (ships with just a noop method).
 module Foreman
   class SilencedLogger < SimpleDelegator
-    def silence(new_level = Logger::ERROR, &block)
-      old_level = self.level
+    def initialize(*args)
+      @@local_levels = Concurrent::Map.new(initial_capacity: 2)
+      super(*args)
+    end
+
+    def local_log_id
+      Thread.current.__id__
+    end
+
```

Comment:
Shouldn't you override the `level` method here to return the `local_level`. How otherwise the logger will even find out what the current level is?